File: mvp/src/main/java/com/stanly/mvp/lce/LceView.java
package com.stanly.mvp.lce;

import android.support.annotation.UiThread;

import com.stanly.mvp.MvpView;

public interface LceView<M> extends MvpView {

    @UiThread
    public void showLoading();

    @UiThread
    public void dismissLoading();

    @UiThread
    public void showContent(M data);

    @UiThread
    public void showError(Throwable e);

    @UiThread
    public void showEmpty();
}

File: mvp/src/main/java/com/stanly/mvp/lce/LceView.kt
package com.stanly.mvp.lce

import android.support.annotation.UiThread

import com.stanly.mvp.MvpView

interface LceView<M> : MvpView {

    @UiThread
    fun showLoading()

    @UiThread
    fun dismissLoading()

    @UiThread
    fun showContent(data: M)

    @UiThread
    fun showError(e: Throwable)

    @UiThread
    fun showEmpty()
}
